File: solutions/ocaml/perfect-numbers/1/perfect_numbers.ml
let classify n = 
  let rec aliquot m sum = 
    if m>n/2 then sum
    else if n mod m = 0 then aliquot (m+1) (sum+m)
    else aliquot (m+1) sum
  in 
  if n<=0 then
    Error "Classification is only possible for positive integers."
  else if n==1 then Ok "deficient"
  else 
    let a = aliquot 1 0
    in 
      if a=n then Ok "perfect"
      else if a<n then Ok "deficient"
      else Ok "abundant"
    
      
    

File: solutions/dart/perfect-numbers/1/lib/perfect_numbers.dart
enum Classification { deficient, abundant, perfect}

class PerfectNumbers {
  
  Classification classify(int num){
    if(num<1){
      throw ArgumentError('Not a positive integer');
    } 
    if(num==1)
      return Classification.deficient;
    int aSum = 1;
    for(var k=2; k< num; k++){
      if(num % k == 0)
        aSum += k;
    }
    if(aSum > num)
      return Classification.abundant;
    else if (aSum == num)
      return Classification.perfect;
    else
      return Classification.deficient;
  }
}
